src/ctap/mod.rs

```diff
@@ -534,11 +534,9 @@ where
         }
         auth_data.extend(vec![0x00, credential_id.len() as u8]);
         auth_data.extend(&credential_id);
-        let cose_key = match pk.to_cose_key() {
-            Some(cose_key) => cose_key,
-            None => return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR),
-        };
-        auth_data.extend(cose_key);
+        if !cbor::write(cbor::Value::Map(CoseKey::from(pk).0), &mut auth_data) {
```

Comment:
Here it should rather be `cbor::write(cbor::Value::from(CoseKey::from(pk)), &mut auth_data)` or `cbor::write(CoseKey::from(pk).into(), &mut auth_data)`, without exposing `CoseKey`'s data publicly.

Reply:
Done